godot-core/src/meta/godot_convert/impls.rs

```diff
@@ -330,6 +331,88 @@ impl FromGodot for u64 {
     }
 }
 
+// ----------------------------------------------------------------------------------------------------------------------------------------------
+// Collections
+
+impl<T: ArrayElement> GodotConvert for Vec<T> {
+    type Via = Array<T>;
+}
+
+impl<T: ArrayElement> ToGodot for Vec<T> {
+    fn to_godot(&self) -> Self::Via {
+        Array::from(self.as_slice())
+    }
+}
+
+impl<T: ArrayElement> FromGodot for Vec<T> {
+    fn try_from_godot(via: Self::Via) -> Result<Self, ConvertError> {
+        Ok(via.iter_shared().collect())
+    }
+}
+
+impl<T: ArrayElement, const LEN: usize> GodotConvert for [T; LEN] {
+    type Via = Array<T>;
+}
+
+impl<T: ArrayElement, const LEN: usize> ToGodot for [T; LEN] {
+    fn to_godot(&self) -> Self::Via {
+        Array::from(self)
+    }
+}
+
+impl<T: ArrayElement, const LEN: usize> FromGodot for [T; LEN] {
+    fn try_from_godot(via: Self::Via) -> Result<Self, ConvertError> {
+        let via_len = via.len(); // Caching this avoids an FFI call
+        if via_len != LEN {
+            let message = format!("Array length mismatch, expected: `{LEN}`, got: `{via_len}`");
+            return Err(ConvertError::with_kind_value(
+                ErrorKind::Custom(Some(message.into())),
+                via,
+            ));
+        }
+
+        let mut option_array = [const { None }; LEN];
+
+        for (idx, element) in via.iter_shared().enumerate() {
+            debug_assert!(
+                idx < LEN,
+                "Elements were added to Array during `iter_shared()`, this is not allowed."
+            );
+
+            // SAFETY: We checked that the length of via matches `LEN`, therefore idx would only be out of bounds if
+            // the array is modified during iteration, which should never be done and if done is caught in debug mode
+            // by the assertion above.
+            unsafe {
```

Comment:
This unsafe is pointless-- the you can simply use iterators, and do `via.iter_shared().zip(&mut option_array)`, and just `*out = element` it like that, or put an `assert!(LEN == via_len)` at the top and index both of them?